public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -161,3 +158,38 @@ export function runPartitionedQuery(datasource: LokiDatasource, request: DataQue
 
   return response;
 }
+
+function getNextRequestPointers(requests: LokiGroupedRequest, requestIndex: number, requestN: number) {
+  // There's a pending metric request:
+  if (requestIndex === 0 && requests[1]?.partition[requestN - 1]) {
+    return {
+      nextRequestIndex: 1,
+      nextRequestN: requestN,
+    };
+  }
+  return {
+    nextRequestIndex: 0,
+    nextRequestN: requestN - 1,
+  };
+}
+
+export function runPartitionedQueries(datasource: LokiDatasource, request: DataQueryRequest<LokiQuery>) {
+  const queries = request.targets.filter((query) => !query.hide);
+  const logQueries = queries.filter((query) => isLogsQuery(query.expr));
+  const metricQueries = queries.filter((query) => !logQueries.includes(query));
```

Comment:
nitpick :) 
you can use `lodash.partition` to get both groups at the same time, something like:
```ts
const [logQueries, metricQueries] = partition(queries, (query) => isLogsQuery(query.expr));
```

Reply:
Oh that's cool. Thanks for the suggestion.

Reply:
Changed.